Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -631,19 +628,8 @@ private static Branch getBranchInstance(@NonNull Context context, boolean isLive
             }
             boolean isNewBranchKeySet;
             if (TextUtils.isEmpty(branchKey)) {
-                // If Branch key is not available check for Fabric provided Branch key
-                String fabricBranchApiKey = null;
-                try {
-                    Resources resources = context.getResources();
-                    fabricBranchApiKey = resources.getString(resources.getIdentifier(FABRIC_BRANCH_API_KEY, "string", context.getPackageName()));
-                } catch (Exception ignore) {
-                }
-                if (!TextUtils.isEmpty(fabricBranchApiKey)) {
-                    isNewBranchKeySet = branchReferral_.prefHelper_.setBranchKey(fabricBranchApiKey);
-                } else {
-                    PrefHelper.Debug("Warning: Please enter your branch_key in your project's Manifest file!");
-                    isNewBranchKeySet = branchReferral_.prefHelper_.setBranchKey(PrefHelper.NO_STRING_VALUE);
-                }
+                PrefHelper.Debug("Warning: Please enter your branch_key in your project's Manifest file!");
+                isNewBranchKeySet = branchReferral_.prefHelper_.setBranchKey(PrefHelper.NO_STRING_VALUE);
```

Comment:
For example, other checks look for a key to start with "key_"

Reply:
> This makes the following check always succeed (i.e. `if (isNewBranchKeySet)`)

I don't think that's the case, because `setBranchKey` returns whether or not a new key has been set. So, if one passes a randomized string every time, `setBranchKey` will return `true` every time, if one passes the same value every time it will return `false`.

I'll go ahead and say that this is the "intended behavior" because Fabric is deprecated, so we expect `fabricBranchApiKey` to always be `null` (if it isn't, that's because someone's _pretending_ to be fabric).

Reply:
Regarding the second comment, I'll push another commit (to this branch), which will standardize how we check for validity of `branchKey`.